unity/ARUI/Assets/Scripts/ARUI/AngelARUI.cs

```diff
@@ -97,24 +99,24 @@ public void SetTasks(string[,] tasks)
     /// <summary>
     /// Set the callback function that is invoked if the user confirms the confirmation dialogue
     /// </summary>
-    public void SetUserIntentCallback(UnityAction userIntentCallBack) => onUserIntentConfirmedAction = userIntentCallBack;
+    public void SetUserIntentCallback(UnityAction<InterpretedAudioUserIntentMsg> userIntentCallBack) => onUserIntentConfirmedAction = userIntentCallBack;
 
     /// <summary>
-    /// If confirmation action is set - SetUserIntentCallback(...) - and no confirmation window is active at the moment, the user is shown a 
-    /// timed confirmation window. Recommended text: "Did you mean ...". If the user confirms the dialogue, the onUserIntentConfirmedAction action is invoked. 
+    /// If confirmation action is set - SetUserIntentCallback(...) - and no confirmation window is active at the moment, the user is shown a
+    /// timed confirmation window. Recommended text: "Did you mean ...". If the user confirms the dialogue, the onUserIntentConfirmedAction action is invoked.
     /// </summary>
     /// <param name="msg">message that is shown in the confirmation dialogue</param>
-    public void TryGetUserFeedbackOnUserIntent(string msg)
+    public void TryGetUserFeedbackOnUserIntent(InterpretedAudioUserIntentMsg intentMsg)
```

Comment:
Ideally, we would modularize the ARUI component from our ROS-specific semantics (e.g. the trans-piled message classes) so that the ARUI could be used by other TA1 members, however if this first demonstration requires this, then we may be able to make due.

A next step may be for there to be a C# class whose properties reflect what our ROS-message has so that we have a 1-to-1 correspondence between the class and the message, but the class is part of the ARUI API and is more transportable.

Reply:
I agree, I would like to keep the ARUI independent (if possible), so it can be used by other systems, but we can leave the changes in the AngelARUI script for the demo.